tools/mksurfdata_esmf/src/mksoiltexMod.F90

```diff
@@ -318,180 +407,26 @@ subroutine mksoiltex(file_mesh_i, file_mapunit_i, file_lookup_i, mesh_o, pioid_o
           ! Determine lookup_index
           lookup_index = mapunit_lookup(mapunit_o(no))
 
-          ! Determine top soil layer sand_o
-          ! If less than 0 search within the SCID array for the first index
-          ! that gives a value greater than or  equal to 0
-          ! Then determine the other soil layers
-          sand_o(no,1) = float(sand_i(1,1,lookup_index))
-          if (sand_o(no,1) < 0._r4) then
-             do l = 2,n_scid
-                if (float(sand_i(1,l,lookup_index)) >= 0._r4) then
-                   sand_o(no,1) = float(sand_i(1,l,lookup_index))
-                   exit
-                end if
-             end do
-          end if
-          if (sand_o(no,1) < 0._r4) then
-             if (int(sand_o(no,1)) == -4) then
-                sand_o(no,:) = 99._r4
```

Comment:
@samsrabin I haven't looked at this in a long while, but judging from `sand_o + clay_o = 100` when these two start equal to -4, I will agree with your assessment that the flag -4 indicates sand dunes regardless of which variable we look at.